spec.md

```diff
@@ -1223,18 +1387,24 @@ compilable to non-parameterized Verilog.
 
 ## Passive Types
 
-It is inappropriate for some circuit components to be declared with a type that
-allows for data to flow in both directions. For example, all sub-elements in a
-memory should flow in the same direction. These components are restricted to
-only have a passive type.
+Stateful elements, such as registers and memories, may contain data of aggregate types.
+Registers with bundle types are especially common.
+However, when using bundle types in stateful elements, the notion of `flip` is meaningless.
+There is no directionality to the data inside a register; the data just *is*.
+
+A passive type is a which does not make use of `flip`.
+More precisely, a passive type is defined recursively:
 
-Intuitively, a passive type is a type where all data flows in the same
-direction, and is defined to be a type that recursively contains no fields with
-flipped orientations. Thus all ground types are passive types. Vector types are
-passive if their element type is passive. And bundle types are passive if no
-fields are flipped and if all field types are passive.
+* All ground types are passive.
+* All probe types are passive.
+* All property types are passive.
+* A vector type is passive if and only if the element type is passive.
+* A bundle type is passive if and only if it contains no field which is marked `flip`
+  and the type of each field is passive.
+* An enum type is passive if and only if the type of each of its variants is passive.
+
+Registers and memories may only be parametrized over passive types.
```

Comment:
Can the PR avoid use of "parameterized". This will eventually mean something else.